src/ui/ui.tsx

```diff
@@ -400,7 +434,17 @@ const App = () => {
         sortMode={sortMode}
       />
       <Sidebar
+        visible={resultGroups.length > 0}
         hiddenEngines={localData.hiddenEngines || []}
+        onToggle={engineId => {
```

Comment:
the `onToggle` method here is the same like below in `<Results`. this needs to be refactored, but I gonna need help for this.